tests/Feature/Repositories/GenreRepositoryTest.php

```diff
@@ -136,6 +137,43 @@ public function testWhenGenreAssociatedWithAlbumItHasManyAlbums()
         $this->assertInstanceOf($this->album->class(), $genre->albums->first());
     }
 
+    /**
+     * Ensure that attempts to create a Genre whose name is identical to an
+     * existing Genre cause an exception to be thrown.
+     *
+     * @return void
+     */
+    public function testWhenGenreNameAlreadyExistsExceptionIsThrown()
+    {
+        $this->repo->create(
+            factory($this->repo->class())->raw(['name' => 'Foo'])
+        );
+
+        try {
+            $this->repo->create(
+                factory($this->repo->class())->raw(['name' => 'Foo'])
+            );
+        } catch (QueryException $e) {
+            $this->assertEquals($e->getCode(), '23000');
```

Comment:
I don't like this because it's fragile, i.e., the test will fail if run against PostgreSQL, for example (because the error code for a unique constraint violation is different.

We've discussed the need for database-agnostic mechanism by which to detect such violations, but have yet to implement one. Maybe now is the time.

Reply:
Another approach is to retrieve all `Genres` whose `name = 'Foo'`, after attempting to create two such records, and ensure that only one is returned. But, as yet, there is no means by which to "find by name" (or any field other than `id`) in our repositories.

Maybe I'm overlooking a simpler way to test this...

Reply:
I don't know if validation would work `["gernre" => "string|unique:genres"]` I think this may do a DB count which I guess would work for both PSQL and MySQL?

Reply:
That's a good idea! Let me give it a whirl...

Reply:
After fiddling with it, validation feels kludgy as compared to detecting and handling a unique constraint violation. I'm going to leave this as-is for now, but implement a mechanism for detecting and handling it for multiple database types in the near future.